jobs/build-arch.Jenkinsfile

```diff
@@ -331,77 +333,14 @@ lock(resource: "build-${params.STREAM}-${basearch}") {
         }
 
-        stage('Kola:QEMU basic') {
-            shwrap("""
-            cosa kola run --rerun --basic-qemu-scenarios --no-test-exit-error
-            cosa shell -- tar -c --xz tmp/kola/ > kola-run-basic.tar.xz
-            cosa shell -- cat tmp/kola/reports/report.json > report.json
-            """)
-            archiveArtifacts "kola-run-basic.tar.xz"
-            if (!pipeutils.checkKolaSuccess("report.json")) {
-                error('Kola:QEMU basic')
-            }
-        }
-
-        // reset for the next batch of independent tasks
-        parallelruns = [:]
-
-        // Kola QEMU tests
-        parallelruns['Kola:QEMU'] = {
-            shwrap("""
-            cosa kola run --rerun --parallel 5 --no-test-exit-error --denylist-test basic  --tag '!reprovision'
-            cosa shell -- tar -c --xz tmp/kola/ > kola-run.tar.xz
-            cosa shell -- cat tmp/kola/reports/report.json > report.json
-            """)
-            archiveArtifacts "kola-run.tar.xz"
-            if (!pipeutils.checkKolaSuccess("report.json")) {
-                error('Kola:QEMU')
-            }
-            shwrap("""
-            cosa shell -- rm -rf tmp/kola
-            cosa kola run --rerun --no-test-exit-error --tag reprovision
-            cosa shell -- tar -c --xz tmp/kola/ > kola-run-reprovision.tar.xz
-            cosa shell -- cat tmp/kola/reports/report.json > report.json
-            """)
-            archiveArtifacts "kola-run-reprovision.tar.xz"
-            if (!pipeutils.checkKolaSuccess("report.json")) {
-                error('Kola:QEMU')
-            }
-        }
-
-        // Kola QEMU Upgrade tests
-        parallelruns['Kola:QEMU Upgrade'] = {
-            // If upgrades are broken `cosa kola --upgrades` might
-            // fail to even find the previous image so we wrap this
-            // in a try/catch so ALLOW_KOLA_UPGRADE_FAILURE can work.
-            try {
-                shwrap("""
-                cosa kola --rerun --upgrades --no-test-exit-error
-                cosa shell -- tar -c --xz tmp/kola-upgrade/ > kola-run-upgrade.tar.xz
-                cosa shell -- cat tmp/kola-upgrade/reports/report.json > report.json
-                """)
-                archiveArtifacts "kola-run-upgrade.tar.xz"
-                if (!pipeutils.checkKolaSuccess("report.json")) {
-                    error('Kola:QEMU Upgrade')
-                }
-            } catch(e) {
-                if (params.ALLOW_KOLA_UPGRADE_FAILURE) {
-                    warnError(message: 'Upgrade Failed') {
-                        error(e.getMessage())
-                    }
-                } else {
-                    throw e
-                }
-            }
-        }
-
-        // process this batch
-        parallel parallelruns
-
+        // Run Kola Tests
+        def n = 4 // VMs are 2G each and arch builders have approx 32G
+        kola(cosaDir: env.WORKSPACE, parallel: n,
+             build: newBuildID, arch: basearch,
```

Comment:
Do we actually need to pass `build` here?

Reply:
No. Resolved